payload/local/local.go

```diff
@@ -0,0 +1,16 @@
+// Local payloads that are used for privilege escalation or non-remote contexts.
+//
+// The local package contains a combination of embed code used for local privilege escalation and
+// allows for the use in other attack patterns. Each payload is documented with what operating
+// system and architecture it supports if necessary.
+package local
+
+type Local interface{}
+
+// Defines default structures and payload functions.
+type (
+	PrivEscLibraryPayload struct{}
```

Comment:
I think my complaint is that PrivEscLib (and privesclib_amd64) are really generic names. Is that... ok? Like I can see a world where we want to do more of these right?

Reply:
I was also quite torn on these, but that part is supposed to be generic. Since that's the type that gets functions defined from: ie the actual payload call in this is `local.PrivEscLibrary.AMD64()`. I think that keeping the top level struct type as broad is fine, but maybe the function signature should be something like `LinuxSharedObjectAMD64()` vs `AMD64()` or even have another type for OS and then keep the architecture?

I'm not sold on any of these as the best way to do it (I'm not even positive embedding binaries is a good idea period), so if you have any thoughts I'm happy to rearrange.

Reply:
I do think that having something with the final signature of `local.PrivEscLibrary.Linux.AMD64()` would probably be fine and cover most the use cases while being specific enough to prevent confusion.... *but* I also hate all the dots lol. We could even do a `local.PrivEscLibrary.LinuxSharedObjectAMD64()`, but I fear the `protocol` signature explosion.

Reply:
> I'm not sold on any of these as the best way to do it (I'm not even positive embedding binaries is a good idea period), so if you have any thoughts I'm happy to rearrange.

:rofl: 

I am also not positive. However, it's an interesting step forward and code can always be removed from the library. I'm not afraid to bump the major version :) I think maybe more experience using this part will inform us on how it should be improved. That's fine by me. Go ahead and merge when you are ready.